pkg/admission/mutator/shoot.go

```diff
@@ -89,70 +86,37 @@ func (s *shoot) Mutate(_ context.Context, newObj, oldObj client.Object) error {
 	overlayDisabled := false
 
 	if shoot.Spec.Networking != nil && shoot.Spec.Networking.Type != nil {
```

Comment:
Do we need to check for `shoot.Spec.Networking.Type != nil`? It is not strictly wrong, but probably not required anymore.